docs/data/data-grid/editing/ValueParserSetterGrid.tsx

```diff
@@ -4,7 +4,6 @@ import {
   GridColDef,
```

Comment:
*(it's unrelated to this file, just adding a comment here to have a separate discussion thread)*

@DanailH 
> I had to keep the following deprecations: GridRenderCellParams['api'], GridRenderEditCellParams['api'], GridValueGetterParams['api'] and GridValueFormatterParams['api'] and GridSortCellParams['api']. The reason is that they are being used internally and It getting access to the apiRef otherwise will be difficult.

Interesting, if removing `api` complicates things, maybe we should reconsider deprecations and remove the deprecation notes in some places?

For `GridRenderCellParams['api']` and `GridRenderEditCellParams['api']` specifically, what is the main challenge?
For example here we use `useGridApiContext`:
https://github.com/mui/mui-x/blob/eb05f2d745b7df086dad9852aa99368426beb56b/packages/grid/x-data-grid/src/components/cell/GridBooleanCell.tsx#L47

For `GridValueGetterParams['api']`, `GridValueFormatterParams['api']` and `GridSortCellParams['api']` I assume the problem is that there's no way to get an `apiRef` in places like this:
https://github.com/mui/mui-x/blob/eb05f2d745b7df086dad9852aa99368426beb56b/packages/grid/x-data-grid/src/colDef/gridCheckboxSelectionColDef.tsx#L25

I don't really see a good solution to this one, so my suggestion would be to keep `api` there and remove the deprecation notes.

Also, what was the reason to deprecate `api` in the first place?
Was it just API ergonomics, or were there some technical aspects behind the decision?

Reply:
Yes, that's the problem - no easy way to get access to the `apiRef`. I think it was deprecated for consistency reasons but I think it makes sense to keep it in some places, basically evaluate it on a case-by-case basis.
I've already removed the deprecation messages on the places I had to keep the `api`.

Reply:
What about GridRenderCellParams['api'] and GridRenderEditCellParams['api'] specifically?
Wouldn't `useGridApiContext` work as replacement?

Reply:
> What about GridRenderCellParams['api'] and GridRenderEditCellParams['api'] specifically? Wouldn't `useGridApiContext` work as replacement?

I tried but I hit a roadblock because `createGroupingColDef.tsx` depends on the `api` to be passed through the `params`.

Reply:
> As an alternative to
> 
> https://github.com/mui/mui-x/blob/eb05f2d745b7df086dad9852aa99368426beb56b/packages/grid/x-data-grid/src/colDef/gridCheckboxSelectionColDef.tsx#L25
> 
> , we could add a 2nd argument to `valueGetter` containing a set of utility functions, where one is a helper to call selectors.
> ```tsx
> const checkboxColumn: GridColDef = {
>     ...GRID_CHECKBOX_SELECTION_COL_DEF,
>     valueGetter: (params: GridValueGetterParams, utils) => {
>         const selectionLookup = utils.invokeSelector(selectedIdsLookupSelector);
>         return selectionLookup[params.id] !== undefined;
>     }
> }
> ```
> 
> This utils object could be obtained from an API method and we pass it in every function from the column definition. It solves the problem of manually passing `api` and also doesn't expose the API.

The issue I see with this approach is how we know which methods are useful and which aren't. If it is just for internal use it would be better to pass in the `api` directly but not document it maybe?

Reply:
> Currently, api is typed with any to avoid having to introduce generics for apiRef. The idea to deprecate it was because it couldn't be typed correctly without big changes. 

We can use `GridApiCommunity` instead of `any` by default:
```diff
diff --git a/packages/grid/x-data-grid/src/models/params/gridCellParams.ts b/packages/grid/x-data
-grid/src/models/params/gridCellParams.ts
index 98f8f3c7d..a9155b805 100644
--- a/packages/grid/x-data-grid/src/models/params/gridCellParams.ts
+++ b/packages/grid/x-data-grid/src/models/params/gridCellParams.ts
@@ -9,6 +9,7 @@ import {
 } from '../gridRows';
 import type { GridStateColDef } from '../colDef/gridColDef';
 import { GridEditCellProps } from '../gridEditRowModel';
+import { GridApiCommunity } from '../api/gridApiCommunity';
 
 /**
  * Object passed as parameter in the column [[GridColDef]] cell renderer.
@@ -129,7 +130,7 @@ export interface GridValueGetterParams<
    * GridApi that let you manipulate the grid.
    * @deprecated Use the `apiRef` returned by `useGridApiContext` or `useGridApiRef` (only available in `@mui/x-data-grid-pro`)
    */
-  api: any;
+  api: GridApiCommunity;
   /**
    * The default value for the cell that the `valueGetter` is overriding.
    */
```

And for accessing Pro/Premium methods `GridApi` type assertion can be used:
```tsx
import { GridApi } from '@mui/x-data-grid-premium';

valueGetter: (params) => {
  const api = params.api as GridApi;
  //     ^--- const api: GridApiPremium
},
```

It's not perfect though.

> @flaviendelangle proposed to pass only state and instanceId in https://github.com/mui/mui-x/pull/4089#discussion_r819564284

Yes, that would work but it won't allow to use methods.

Reply:
Ok, so as far as I see we have 2 main things that we need to decide on:
1. Do we keep the `api` method in these cases? I would vote to keep them because it would be difficult to remove them as they are used internally. I would also be in favor of keeping the entinre `api` and not having a substep of utility methods. If we go with this approach then the next question would be doing we add the `api` to all the callbacks in the `GridColDef`. It makes sense to provide it so we make it easier for people to do global column definitions.
2. The typing - if the `GridApiCommunity` works we can go with this approach.

Reply:
Updated